dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java

```diff
@@ -267,14 +271,21 @@ private WorkflowVersion updateDBVersionSourceFilesWithRemoteVersionSourceFiles(W
      * - Delete version for corresponding service and workflow
      * @param repository Repository path (ex. dockstore/dockstore-ui2)
      * @param gitReference Git reference from GitHub (ex. refs/tags/1.0)
+     * @param username Git user who triggered the event
+     * @param installationId GitHub App installation ID
      * @return List of updated workflows
      */
-    protected List<Workflow> githubWebhookDelete(String repository, String gitReference) {
+    protected List<Workflow> githubWebhookDelete(String repository, String gitReference, String username, String installationId) {
         // Retrieve name from gitReference
         Optional<String> gitReferenceName = GitHelper.parseGitHubReference(gitReference);
         if (gitReferenceName.isEmpty()) {
             String msg = "Reference " + gitReference + " is not of the valid form";
             LOG.error(msg);
+            LambdaEvent lambdaEvent = createBasicEvent(repository, gitReference, username, installationId, LambdaEvent.LambdaEventType.DELETE);
+            lambdaEvent.setMessage(msg);
+            lambdaEvent.setSuccess(false);
+            lambdaEventDAO.create(lambdaEvent);
+            sessionFactory.getCurrentSession().getTransaction().commit();
```

Comment:
A little wary of this in production code, why is this necessary and has this pattern been used before?

Reply:
Since an error is being thrown, the transaction is rolled back, so the message would never be written. We have used this once before for Zenodo I believe.

Reply:
What if I clear the session before updating(sessionFactory.getCurrentSession().clear()). From what I understand this would mean that no unexpected things would be saved, only the lambda event.

Reply:
The docs say that a clear 

> ... cancel all pending saves, updates and deletions.

Reply:
That does sound a bit safer